applications/StructuralMechanicsApplication/custom_utilities/structural_mechanics_math_utilities.hpp

```diff
@@ -373,6 +373,64 @@ class StructuralMechanicsMathUtilities
             return ReferencePosition[0];
         }
     }
+
+    /**
+     * Transforms a 2-point tensor from an origin system to a target system
+     * M=M_ij origin_left x origin_right = M_lk target_left X target_right
+     * @return rOriginLeft: matrix with the basis vectors of basis origin left as columns
+     * @return rOriginRight: matrix with the basis vectors of basis origin right as columns
+     * @return rTargetLeft: matrix with the basis vectors of basis target left as columns
+     * @return rTsargetRight: matrix with the basis vectors of basis target right as columns
+     * @return rTensor: the tensor to be tranformed
+     */
+    template<int TDim>
+    static inline void TensorTransformation(
+        bounded_matrix<double,TDim,TDim>& rOriginLeft,
+        bounded_matrix<double,TDim,TDim>& rOriginRight,
+        bounded_matrix<double,TDim,TDim>& rTargetLeft,
+        bounded_matrix<double,TDim,TDim>& rTargetRight,
+        bounded_matrix<double,TDim,TDim>& rTensor)
+    {
+        // metric computation (of the target systems)
+        bounded_matrix<double,TDim,TDim> metric_left = ZeroMatrix(TDim,TDim);
+        bounded_matrix<double,TDim,TDim> metric_right = ZeroMatrix(TDim,TDim);
+        for(int i=0;i<TDim;i++){
+            for(int j=0;j<TDim;j++){
+                metric_left(i,j) += inner_prod(column(rTargetLeft,i),column(rTargetLeft,j));
+                metric_right(i,j) += inner_prod(column(rTargetRight,i),column(rTargetRight,j));
+            }        
+        }
+
+        // invert metric
+        double det;
+        Matrix inv_metric_left = Matrix(TDim,TDim);
```

Comment:
Please use bounded matrix if you know the size

Reply:
done